forge/ee/routes/catalogues/index.js

```diff
@@ -62,6 +65,92 @@ module.exports = async function (app) {
         }
     })
 
+    /**
+     * Uploads a NPM packaged subflow to teams private repo
+     * @name /api/v1/teams/:teamId/npm/subflow
+     * @static
+     * @memberof forge.routes.api.team.npm
+     */
+    app.put('/npm/subflow', {
+        preHandler: async (request, reply) => {
+            if (request.session.ownerType !== 'project') {
+                reply.code(401).send({ code: 'unauthorized', error: 'unauthorized' })
+            }
+            const project = await app.db.models.Project.byId(request.session.ownerId)
+            if (request.params.teamId !== project.Team.hashid) {
+                // AccesToken being used - but not owned by project in this team
+                reply.code(404).send({ code: 'not_found', error: 'Not Found' })
+                return // eslint-disable-line no-useless-return
+            }
+        },
+        schema: {
+            summary: 'Allows Subflow packages to be stored in the Team Registry',
+            tags: ['NPM Package'],
+            body: {
+                type: 'object',
+                properties: {
+                    package: {
+                        type: 'object',
+                        additionalProperties: true
+                    },
+                    subflow: {
+                        type: 'object',
+                        additionalProperties: true
+                    }
+                }
+            },
+            params: {
+                type: 'object',
+                properties: {
+                    teamId: { type: 'string' }
+                }
+            },
+            response: {
+                200: {
+                    type: 'object'
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        const package = request.body.package
+        const subflowJSON = request.body.subflow
+        if (!package || !subflowJSON) {
+            reply.status(422).send({ error: 'missing_values', message: 'Missing values' })
+            return
+        }
+        if (!package.name.startsWith(`@flowfuse-${request.team.hashid}/`)) {
+            reply.status(403).send({ error: 'not_authorized', message: 'Not Authorized' })
+            return
+        }
+        if (!package.version || !semver.valid(package.version)) {
+            reply.status(422).send({ error: 'bad_version', message: 'Invalid semver' })
+            return
+        }
```

Comment:
```suggestion
        if (!semver.valid(package.version)) {
            reply.status(422).send({ error: 'bad_version', message: 'Invalid semver' })
            return
        }
```

Let swagger take care of missing props

Reply:
Same again 422 is the better status code